File: backend/app/models/task.py
from __future__ import annotations
from datetime import datetime
from typing import Optional, List, TYPE_CHECKING

from sqlmodel import Field, SQLModel

if TYPE_CHECKING:
    from app.models.user import User


class TaskBase(SQLModel):
    title: str = Field(min_length=1, max_length=255, index=True)
    description: Optional[str] = Field(default=None, max_length=1000)
    completed: bool = Field(default=False)

    owner_id: Optional[int] = Field(default=None, foreign_key="user.id")

class Task(TaskBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    created_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
    updated_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)



class TaskCreate(TaskBase):
    pass

class TaskRead(TaskBase):
    id: int
    created_at: datetime
    updated_at: datetime

class TaskUpdate(SQLModel):
    title: Optional[str] = Field(default=None, min_length=1, max_length=255)
    description: Optional[str] = Field(default=None, max_length=1000)
    completed: Optional[bool] = Field(default=None)

File: frontend/src/types/task.ts
// frontend/src/types/task.ts

export interface Task {
  id: number;
  title: string;
  description: string | null;
  status: 'pending' | 'completed';
  owner_id: number;
}

export interface TaskCreate {
  title: string;
  description?: string;
  status?: 'pending' | 'completed';
}

export interface TaskUpdate {
  title?: string;
  description?: string | null;
  status?: 'pending' | 'completed';
}